src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,107 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    private static class Node<K, V> {
+        final K key;
+        V value;
+        Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public final String toString() {
+            return "key = " + key + ", value = " + value;
+        }
+
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = getIndex(key);
+        Node<K, V> temp = table[hash];
+        Node<K, V> current = temp;
+        if (temp == null) {
+            table[hash] = new Node<K, V>(hash, key, value, null);
+        } else {
+            while (temp != null) {
+                if (Objects.equals(temp.key, key)) {
+                    temp.value = value;
+                    return;
+                }
+                current = temp;
+                temp = temp.next;
+            }
+            current.next = new Node<K, V>(hash, key, value, null);
+        }
+        size++;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getIndex(key);
+        Node<K, V> temp = table[hash];
+        while (temp != null) {
+            if (Objects.equals(temp.key, key)) {
+                return temp.value;
+            }
+            temp = temp.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void transferValue(K key, V value, Node<K, V>[] table) {
+        int hash = getIndex(key);
+        Node<K, V> temp = table[hash];
+        if (temp == null) {
+            table[hash] = new Node<K, V>(hash, key, value, null);
```

Comment:
```suggestion
            table[hash] = new Node<>(hash, key, value, null);
```
the same everywhere)